domain/src/main/java/com/github/dodobest/domain/Calculator.kt

```diff
@@ -0,0 +1,186 @@
+        return inputString.replace(" ", "")
+    }
+
+    private fun addNumAndSingToArrayFromString(inputString: String, numAndSignArray: MutableList<String>) {
+        for (idx in inputString.indices) {
+            // 문자열로 부터 숫자와 문자를 분리하기
+            splitNumAndSignFromString(inputString, idx, numAndSignArray)
+        }
+
+        // 마지막 숫자 값 더해주고, 정적 변수 초기화 하기
+        numAndSignArray.add(inputString.slice(IntRange(firstNumIndex, inputString.length-1)))
+
+        firstNumIndex = -1
+        isMinusSign = false
+
+    }
+
+    private fun splitNumAndSignFromString(inputString: String, charIndex: Int, numAndSignArray: MutableList<String>) {
+        // 숫자인 경우
+        if (checkCharIsNum(inputString[charIndex])) {
+            // 숫자를 나타내는 처음 Index 값 업데이트 하기
+            updateFirstNumIndex(inputString, charIndex)
+            return
+        }
+
+        // 연산 기호인 경우
+        checkArithmeticOperation(inputString, charIndex, numAndSignArray)
+    }
+
+    private fun calculate(inputArray: List<String>): Double {
+        var sum: Double = inputArray[0].toDouble()
+        var idx = 1
+
+        while (idx < inputArray.size - 1) {
+            val num: Double = inputArray[idx+1].toDouble()
+
+            sum = calcWithOperation(sum, inputArray[idx], num)
+            idx += 2
+        }
+        return sum
+    }
+
+    private fun calcWithOperation(sum: Double, operation: String, num: Double): Double = when (operation) {
+        PLUS.toString() -> sum + num
+        MINUS.toString() -> sum - num
+        MULTIPLY.toString() -> sum * num
+        DIVIDE.toString() -> sum / num
+        else -> {
+            throw IllegalArgumentException("사칙연산 외 기호가 입력되었습니다.")
+        }
+    }
+
+    private fun updateFirstNumIndex(inputString: String, charIndex: Int){
+        // 값을 나타내는 처음 숫자가 이미 있다면 업데이트 하지 않음
+        if (firstNumIndex != -1) {
+            return
+        }
+
+        // 값 '0'이 아닌 0으로 시작하는 숫자를 입력한 경우 예외 처리
+        if (checkNumStartWithZeroAndNotExactZero(inputString, charIndex)) {
+            throw IllegalArgumentException("0으로 시작하는 숫자는 지원하지 않습니다.")
+        }
+        firstNumIndex = charIndex
+    }
+
+    private fun checkCharIsNum(charVal: Char): Boolean {
+        if (charVal.code in '0'.code..'9'.code ) {
+            return true
+        }
+        return false
+    }
+
+    private fun checkNumStartWithZeroAndNotExactZero(inputString: String, charIndex: Int): Boolean {
+        if (inputString[charIndex] == '0' && charIndex < inputString.length - 1 && checkCharIsNum(inputString[charIndex + 1])) {
+            return true
+        }
+        return false
+    }
+
+    private fun checkArithmeticOperation(inputString: String, charIndex: Int, numAndSignArray: MutableList<String>) {
+        val inputChar: Char = inputString[charIndex]
+
+        // 사칙 연산 기호가 아닌 경우 IllegalArgumentException throw
+        if (!charIsOperation(inputChar)) {
+            throw IllegalArgumentException("사칙 연산 외 기호가 입력되었습니다.")
+        }
```

Comment:
```suggestion
    require(charIsOperation(inputChar)) { "사칙 연산 외 기호가 입력되었습니다." }
```
코틀린의 require를 활용해볼 수 있습니다.